app/src/gui/components/notebook/record_table.tsx

```diff
@@ -72,6 +77,11 @@ type RecordsBrowseTableProps = {
 function RecordsTable(props: RecordsTableProps) {
   const {project_id, maxRows, rows, loading} = props;
 
+  const {data: uiSpec} = useQuery({
+    queryKey: ['uiSpec', project_id],
+    queryFn: () => getUiSpecForProject(project_id),
+  });
```

Comment:
I think this could be helpful in other areas so would prefer you make a simple custom hook e.g. useGetUiSpec which wraps the useQuery.